src/web/src/components/Search/Response.js

```diff
@@ -28,13 +28,17 @@ class Response extends Component {
     state = { 
         tree: buildTree(this.props.response), 
         downloadRequest: undefined, 
-        downloadError: '' 
+        downloadError: '',
+        isFolded: this.props.isInitiallyFolded,
     }
 
     componentDidUpdate = (prevProps) => {
         if (JSON.stringify(this.props.response) !== JSON.stringify(prevProps.response)) {
             this.setState({ tree: buildTree(this.props.response) });
         }
+        if (this.props.isInitiallyFolded !== prevProps.isInitiallyFolded) {
+            this.setState({ isFolded: this.props.isInitiallyFolded });
+        }
```

Comment:
this is the correct pattern when using class-based components and lifecycle functions.  functional components are the way moving forward; anything new should be functional, and older class-based components like this will eventually be rewritten.

i'm only mentioning this since you said you were new to react, and i thought it was worth clarifying.  couple of links, if you are interested:

https://reactjs.org/docs/components-and-props.html
https://www.pluralsight.com/guides/prop-changes-in-react-component